packages/benchmarks/lit-element/class-map/index.ts

```diff
@@ -0,0 +1,235 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {html, LitElement} from 'lit-element';
+import {repeat} from 'lit-html/directives/repeat.js';
+import {classMap} from 'lit-html/directives/class-map.js';
+import {queryParams} from '../../utils/query-params.js';
+
+/**
+ * This benchmark renders templates, updates templates and nop-updates.
+ *
+ * In each of these benchmarks we measure between 2 variants of classmaps,
+ * and raw string manipulation.
+ *
+ * The benchmark itself creates a list of elements, and then selects one of
+ * them.
+ */
+
+interface IData {
+  id: number;
+  label: string;
+}
+
+enum DirectiveVariant {
+  // This uses the imported classMap directive.
+  CurrentImplementation,
+  // This uses a stripped down classMap directive.
+  SimpleImplementation,
+  // No directive, only string concatenation.
+  StringConcatenation,
+}
+
+function simpleClassMap(classInfo: Record<string, boolean>) {
+  // Render implementation from classMap directive.
+  return Object.keys(classInfo)
```

Comment:
Add a space before and after the string to keep classes from merging with strings outside of the expression. ie: `class="one${classMap(...)}two"`